pkg/client/client.go

```diff
@@ -68,7 +69,7 @@ func NewStreamClient(gateway string, topic string, acceptableContentType string)
 
 func (lc *StreamClient) Publish(ctx context.Context, payload io.Reader, key io.Reader, contentType string, headers map[string]string) (PublishResult, error) {
 	m := serialization.Message{}
-	if lc.acceptableContentType != contentType { // TODO support smarter compatibility (eg subtypes)
+	if !strings.HasPrefix(contentType, lc.acceptableContentType) { // TODO support smarter compatibility (eg subtypes)
```

Comment:
This is not the proper way to support subtypes.
I'd rather leave it as is for now, unless this is blocking you (it shouldn't)

Reply:
I am using RestTemplate to post request from function. RestTemplate uses `application/json;utf-8` by default. I didn't want to change it there as it made the function looks complicated.

Reply:
At the very least, chop until `;` on both sides and continue to do an equality check

Reply:
done!